src/constants.ts

```diff
@@ -4,20 +4,111 @@ export const SUPPORTED_CHAINS = Object.values(Network) as Network[];
 export const DEFAULT_PRIVATE_KEY =
   '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80'; // hardhat 0 address
 
-// @dev if the script is being run via the mdev command, we are in dev mode
-export const IS_DEV = process.argv[1].includes('mdev');
+export const CONNECTIONS = {
+  prod: {
+    metalWeb: 'https://metal.build',
+    metalService: 'https://api.metal.build',
+    auth0: {
+      issuer: 'https://auth.metal.build',
+      clientId: '9TFnIsSYlxiSKIs5bvwhfxu9yQFvhT0R',
+      audience: 'https://api.metal.build',
+    },
+  },
+  staging: {
+    metalWeb: 'https://staging.metal.build',
+    metalService: 'https://staging.api.metal.build',
+    auth0: {
+      issuer: 'https://metal-build-dev.us.auth0.com',
+      clientId: 'KvzTMmpjygjJTTlwfNNv1d4k0Xo93MUW',
+      audience: 'https://staging.api.metal.build',
+    },
+  },
+  dev: {
+    metalWeb: 'http://localhost:3000',
+    metalService: 'http://localhost:1234',
+    /**
+     * @dev the issuer, clientId, and audience are all the same for dev and staging
+     */
+    auth0: {
+      issuer: 'https://metal-build-dev.us.auth0.com',
+      clientId: 'KvzTMmpjygjJTTlwfNNv1d4k0Xo93MUW',
+      audience: 'https://staging.api.metal.build',
+    },
+  },
+};
 
-export const PREVIEW_SERVICE_URL = process.env.PREVIEW_SERVICE_URL
-  ? process.env.PREVIEW_SERVICE_URL
-  : IS_DEV
-  ? 'http://localhost:1234'
-  : 'https://api.metal.build';
+// @dev the config for the cli is determined by the following order of preference:
+//  1. scoped flags (--metal-service, --metal-web, --auth0-issuer, --auth0-client-id, --auth0-audience)
+//  2. env vars (METAL_SERVICE, METAL_WEB, AUTH0_[ISSUER, CLIENT_ID, AUDIENCE])
+//  3. MODE flags (--staging, --dev, --prod)
+const bootstrapConstants = () => {
+  const flagIsPresent = (flag: string): boolean => process.argv.some(arg => arg === flag);
+  const getFlagValueFromArgv = (flag: string): string | undefined => {
+    const flagIndex = process.argv.findIndex(arg => arg === flag);
+    if (flagIndex === -1) return undefined;
+    else return process.argv[flagIndex + 1];
+  };
 
-export const PREVIEW_WEB_URL = process.env.PREVIEW_WEB_URL
-  ? process.env.PREVIEW_WEB_URL
-  : IS_DEV
-  ? 'http://localhost:3000'
-  : 'https://metal.build';
+  const env = process.env;
+
+  const MODE: 'dev' | 'staging' | 'prod' = flagIsPresent('--staging')
+    ? 'staging'
+    : flagIsPresent('--dev') || (process.argv[1].endsWith('mdev') && !flagIsPresent('--prod'))
+    ? 'dev'
+    : 'prod';
```

Comment:
🎏 I like these flags! not blocking, but we might also consider throwing if more than one is passed. Something like: 
```
function maxOneFlag(...flags: string[]): boolean {
  return flags.filter(flagIsPresent).length <= 1;
}

if (!maxOneFlag('--staging', '--prod', '--dev')) {
  throw new Error('Only one of --staging, --prod, or --dev should be passed');
}
```
not sure if that belongs here or in some initial run validation.

Reply:
will do